code/Calendar.php

```diff
@@ -276,19 +276,31 @@ public static function getWeekdayName($value, $width = 'wide', $locale = '', $st
     {
         $result = '';
         if ((!empty($value)) || ($value === 0) || ($value === '0')) {
-            $weekday = null;
-            if (is_int($value)) {
-                $weekday = $value;
-            } elseif (is_float($value)) {
-                $weekday = intval($value);
-            } elseif (is_string($value)) {
-                if (is_numeric($value)) {
+            switch (gettype($value)) {
```

Comment:
At least the german documentation states this: http://php.net/manual/de/function.gettype.php

> Benutzen Sie gettype() niemals, um auf einen bestimmten Typ zu prüfen, da die zurückgegebene Zeichenkette irgendwann geändert werden könnte. Außerdem ist diese Methode langsam, da sie einen Vergleich von Zeichenketten benutzt.

Very rough translation: "Never use `gettype()` because the returned values may change in the future"

However, the english documentation only has a "For type checking, use is_* functions" hint… In the past it had that hint as well: https://web.archive.org/web/20110717210348/http://www.php.net/manual/en/function.gettype.php

So not sure… ;-)